api.swagger.yaml

```diff
@@ -467,96 +576,803 @@ paths:
+        '404':
+          description: The UUID does not correspond to an existing group.
+      description: Get a list of group posts.  
+  /groups/search:
+    get:
+      tags:
+        - groups
+      parameters:
+        - description: Search term.
+          in: query
+          name: fulltext_search
+          schema:
+            type: string
+        - description: Filter by UUID of a category.
+          in: query
+          name: category
+          schema:
+            type: string
+        - $ref: '#/components/parameters/offsetParam'
+        - $ref: '#/components/parameters/limitParam'
+      responses:
+        '200':
+          description: List of groups matching the filter criteria.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/group-search-schema'
+        '404':
+          description: No matching groups were found.
+    description: Search for groups. Returns group snippets.
+  /groups/categories:
+    get:
+      tags:
+        - groups
+      responses:
+        '200':
+          description: List of group categories.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/title-uuid-list-common-schema'
+        '404':
+          description: No group categories were found.
+    description: List all group categories.
+  '/groups/{group_id}':
+    get:
+      tags:
+        - groups
+      parameters:
+        - description: The Invotra UUID of the group.
+          in: path
+          name: group_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: Returns a single complete group object.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/group-schema'
+        '400':
+          description: The supplied UUID was malformed.
+        '404':
+          description: The UUID does not correspond to an existing group.
+      description: Get information about the group.
+  '/groups/{group_id}/members':
+    get:
+      tags:
+        - groups
+      parameters:
+        - description: The Invotra UUID of the group.
+          in: path
+          name: group_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: |
+            List of group members. If there are no members, an empty array.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/group-membership-list-schema'
```

Comment:
can we pull out the array to avoid the null bug

Reply:
updated